mobile/src/screens/chat/chat.tsx

```diff
@@ -101,16 +102,19 @@ const Chat: React.FC = () => {
       </View>
       <ScrollView style={styles.chatWrapper} ref={scrollViewReference}>
         {currentChatMessages.map((item, index) => {
-          const previousMessage = currentChatMessages[index - PREVIOUS_USER];
-          const isDifferentMessageOwner =
-            item.senderId !== previousMessage?.senderId ||
-            item.chatId !== previousMessage.chatId;
+          const { isTimeVisible, isDifferentMessageOwner, currentTime } =
+            messageItemProperties({
+              messages: currentChatMessages,
+              currentIndex: index,
+            });
 
           return (
             <MessageItem
               text={item.message}
+              time={currentTime ?? '00:00'}
+              isTimeVisible={isTimeVisible}
               isUser={item.senderId === authenticatedUser.id}
-              isAvatarVisible={isDifferentMessageOwner}
+              isAvatarVisible={isDifferentMessageOwner ?? true}
```

Comment:
Is `isDifferentMessageOwner` a `boolean | undefined`? In any case `?? true` is quite hard to understand from the first view